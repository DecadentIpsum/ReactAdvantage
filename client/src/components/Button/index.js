import React, { Component } from 'react';
import { Button as PrimeButton } from 'primereact/components/button/Button';

export default class Button extends Component {
    render() {
        const { secondary, danger, children, ...restProps } = this.props;
        const classNames = [];
        secondary && classNames.push('p-button-secondary');
        danger && classNames.push('p-button-danger')

        return (
            <PrimeButton
                className={classNames.join(' ')}
                {...restProps}
            />
        );
    }
}
