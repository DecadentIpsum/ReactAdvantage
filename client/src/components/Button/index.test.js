import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Button from './index';

test('default button renders correctly', () => {
    const component = shallow(<Button />);

    expect(component).toMatchSnapshot();
});

test('secondary button renders correctly', () => {
    const component = shallow(<Button secondary />);

    expect(component).toMatchSnapshot();
});

