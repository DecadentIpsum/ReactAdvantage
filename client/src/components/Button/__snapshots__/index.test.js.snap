// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default button renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "",
      "cornerStyleClass": "ui-corner-all",
      "icon": null,
      "iconPos": "left",
      "label": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "",
        "cornerStyleClass": "ui-corner-all",
        "icon": null,
        "iconPos": "left",
        "label": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`secondary button renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    secondary={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "ui-button-secondary",
      "cornerStyleClass": "ui-corner-all",
      "icon": null,
      "iconPos": "left",
      "label": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "ui-button-secondary",
        "cornerStyleClass": "ui-corner-all",
        "icon": null,
        "iconPos": "left",
        "label": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
